src/main/main.ts

```diff
@@ -34,8 +39,87 @@ export function importOnestopToBasecamp(): void {
     deleteOldRows(processedRowIds);
 }
 
+/**
+ * Processes existing event rows by checking whether the row has been modified and if so indiscriminately
+ * updates all the tasks with the new information
+ * 
+ * @param row 
+ */
 function processExistingRow(row: Row): void {
-    // Will be implemented as part of https://3.basecamp.com/4474129/buckets/38736474/todos/7717428992
+
+    Logger.log("Checking if the row has changed...\n")
+
+    if(hasChanged(row)) {
+
+        Logger.log("Changes detected in row! Updateding associated todos to reflect changes...\n")
+        
+        const savedRowBasecampMapping: RowBasecampMapping | null = getRowBasecampMapping(row);
+        
+        if(savedRowBasecampMapping == null) {
+            throw new RowBasecampMappingMissingError("The rowBasecampMapping object is null! Unable to proceed with updating the todo!");
+        }
+        
+        const basecampTodoRequests: Map<string, BasecampTodoRequest> = getBasecampTodoRequestsForRow(row);
+        const currentRoles: string[] = Array.from(basecampTodoRequests.keys());
+
+        const roleTodoIdMap = savedRowBasecampMapping.roleTodoIdMap
+        const originalRoles: string[] = Object.keys(roleTodoIdMap)
+
+        for (const role of currentRoles) {
+
+            let todoId = roleTodoIdMap[role];
+            let request = basecampTodoRequests.get(role);
+
+            if(request == undefined) {
+                throw new BasecampRequestMissingError("Missing basecamp request!");
+                
+            } else if (todoId == undefined) {
+                let newTodoId = createTodo(request, DEFAULT_TODOLIST_IDENTIFIER)
+                roleTodoIdMap[role] = newTodoId // Modify the roleTodoIdMap to reflect changes in the document properties
+                
+            } else {
+                
+                let todoIdentifier: TodoIdentifier = {
+                    projectId: PROJECT_ID,
+                    todoId: todoId
+                }
+                
+                Logger.log(`Updating todo for ${row.what.value} (${row.startTime})...\n`)
+                updateTodo(request, todoIdentifier)
+            }
+        }   
+
+        deleteObsoleteTodos(originalRoles, currentRoles, roleTodoIdMap);
+
+        saveRow(row, roleTodoIdMap);
+    }
+}
+
+/**
+ * Deletes todos associated with helper groups that are no longer present on the event row.
+ * The first line checks whether there are any rows present in the original roles that are not present in the current roles (indicating an obsolete role)
+ * 
+ * @param originalRoles a list of all the original roles associated with the row including the lead role 
+ * @param currentRoles a list of all the current roles associated with the row including the lead role. This may be identical to the original roles
+ */
+function deleteObsoleteTodos(originalRoles: string[], currentRoles: string[], roleTodoIdMap: { [key: string]: string }): void {
```

Comment:
this function can be broken into different parts:
1. getObsoleteRoles: takes original and current roles, returns list of obsolete roles
2. getObsoleteTodos: takes list of obsolete roles and roleTodoIdMap, returns todoIds
3. deleteObsoleteTodos: takes todoIds, deletes them
4. removeDeletedRoles: takes roleTodoIdMap, list of obsolete roles, returns a copy of the map with the roles deleted

Reply:
Sounds good, the new code reflects this change request as well